libs/shared/src/hooks/use-staking-actions/use-staking-actions.ts

```diff
@@ -366,15 +645,69 @@ export function useStakingActions() {
       haqqAddress,
       haqqChain,
       getFee,
-      getRedelegationParams,
       signTransaction,
       broadcastTransaction,
+      chainId,
+      posthog,
       getTransactionStatus,
     ],
   );
 
+  // Handle precompile claiming rewards from all validators
+  const handlePrecompileClaimAllRewards = useCallback(async () => {
+    if (!ethAddress || !writeContractAsync) {
+      throw new Error(
+        'Insufficient data for claiming all rewards or simulation error',
+      );
+    }
+
+    const txHash = await writeContractAsync({
+      address: STAKING_PRECOMPILE_ADDRESS,
```

Comment:
Incorrect precompile address. Rewards are being handled by Distribution precompile.
ABI for claiming all rewards
```
{
    "inputs": [
      {
        "internalType": "address",
        "name": "delegatorAddress",
        "type": "address"
      },
      {
        "internalType": "uint32",
        "name": "maxRetrieve",
        "type": "uint32"
      }
    ],
    "name": "claimRewards",
    "outputs": [
      {
        "internalType": "bool",
        "name": "success",
        "type": "bool"
      }
    ],
    "stateMutability": "nonpayable",
    "type": "function"
  },
```